### ISSUE TITLE
Publication of the standard name table XML schema file on the website

### ISSUE THREAD

**Issue by larsbarring** [cf-convention/cf-convention.github.io#469]
This is one in a string of issues that aims to improve the format of the XML version of the standard name table files, see https://github.com/cf-convention/cf-convention.github.io/issues/457 for background and overview.

This particular issue directly builds on #459 , which implements a unified version of the XML standard name table format, as specified in [Appendix B](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#standard-name-table-format), and which is applicable to all published version of the XML file. 

To comply with the XML syntax the schema file (.xsd), if present at all, needs to be specified in a way that is reachable for the software consuming XML file. Currently this is not the case, and the best way to do this is to publish the file on the web site. Thus, the proposal is to add a new line under the heading "**Documents**" on the [Vocabularies page](https://cfconventions.org/vocabularies.html). 

Currently it reads: 
> **Documents**
[Guidelines](https://cfconventions.org/Data/cf-standard-names/docs/guidelines.html) for Construction of CF Standard Names
List of[ contributors](https://cfconventions.org/Data/cf-standard-names/docs/standard-name-contributors.html) to CF Standard Names
[Rules][vocabrules] for making changes to the CF vocabularies

Proposal:
>**Documents**
[Guidelines](https://cfconventions.org/Data/cf-standard-names/docs/guidelines.html) for Construction of CF Standard Names
List of[ contributors](https://cfconventions.org/Data/cf-standard-names/docs/standard-name-contributors.html) to CF Standard Names
[Rules][vocabrules] for making changes to the CF vocabularies
**[Schema file](https://cfconventions.org/Data/schema-files/cf-standard-name-table-2.0.xsd) for the Standard name table XML files (_version 1 -- current_)** 

The last line is bold to only highlight the addition. This addition will make it possible to provide a robust link to the the schema file from within the XML standard name table files, which will be dealt with in a separate issue.

**Comment by larsbarring** [comment]
The associated draft PR (https://github.com/cf-convention/cf-convention.github.io/pull/472) should be activated only after PR https://github.com/cf-convention/cf-convention.github.io/pull/468 has been merge because in that PR the directory structure is established.

But the actual change to the layout of the ***vocabularies web page**, which is in focus for this issue is independent of where the link actually points to.

**Comment by JonathanGregory** [comment]
Thanks, @larsbarring. That makes sense. I support this change.

The text of the link says it's version 1, but the link itself is to your new version 2.0.

Would it be in order for you to fix the line above as well, which is rendered as "[Rules][vocabrules]"? Presumably it should be "Rules" as HTML link text.

**Comment by larsbarring** [comment]
Regarding your first comment I think you mean "*(version 1 -- current)*". This is as intended because this new schema file should apply to *all versions* until we decide to change to the XML format, e.g. by removing AMIP and GRIB, where I totally agree with Alison in that it should be handled later and as a totally independent issue (that this string of issues nevertheless paves the way for).  It also means that *all* published XML files should be processed to include this new link (this is central to this whole effort).

Regarding your second comment, I saw the strange "link", but did not know what to make of it. I also saw a couple of other link issues on the same page; the version 83 HTML and XML links actually point to version 82, and for the KWIK links there are duplicates for version 82. In fact I would prefer not to deal with those issues: 

Firstly, I would like to focus on this rather complex string of issues and associated PRs.

Secondly the other link issues on the vocabularies page could (should preferably, I would say) be handled as part of #408, which again connects to the standard name publication process, but from a totally different angle than what is dealt with here.

**Comment by JonathanGregory** [comment]
Dear @larsbarring 

> Regarding your first comment I think you mean "_(version 1 -- current)_". This is as intended because this new schema file should apply to _all versions_ until we decide to change to the XML format.

Yes, that's what I meant, and I agree it should. But the new scheme file is version 2.0, isn't it, not "version 1 -- current"?

Yes, it's fine to leave the smaller errors on the vocabulary page to a separate issue.

https://github.com/cf-convention/cf-convention.github.io/issues/408 is about adding a link to https://cfconventions.org/Data/cf-standard-names/current/src/cf-standard-name-table.xml, if that intended always to be a link to latest xml. Do you think it is? I can do that PR myself, if that's correct, since it was my issue.

Best wishes

Jonathan

**Comment by larsbarring** [comment]
Thanks Jonathan @JonathanGregory,

If I change the wording to

> [Schema file](https://cfconventions.org/Data/schema-files/cf-standard-name-table-2.0.xsd) for all versions (*1 -- current*) of the Standard name table XML files.

is that clearer? Or do you think that we explicitly need to mention the version of the schema file as such? The mockup link points to the right (*) place and file name. If someone wants to download the file it will be the right version (2.0). Otherwise, the (main) aim with this link is to provide a robust uri that is included in all XML files.

I have not touched the old/present version (1.3) of the schema files that I believe are present for all version of the standard name table as `Data/cf-standard-names/$version$/xsl/html/cf-standard-name-table-1.3.xsl`.

(*) If #433 is accepted without changes.

**Comment by JonathanGregory** [comment]
Sorry, @larsbarring, I had misunderstood "(1 -- current)". I thought it meant that "1" _is_ the current version, with "--" as a dash, rather than indicating a numerical range, which makes more sense! Probably others wouldn't make this mistake; that was just unlucky. On the other hand, you could omit "(1 -- current)". I would not have misunderstood "Schema file for all versions". Jonathan

**Comment by larsbarring** [comment]
Dear Jonathan,
No worries -- thanks for your input :-) 
PR has been updated.
Lars

**Comment by larsbarring** [comment]
With the comments above I think that we can start the countdown clock, which means that the associated PR can be merged **after #468**, but at the earliest on April, 17.

**Comment by larsbarring** [comment]
Hi @sadielbartholomew,

May I again make use of you expertise to have a look at the associated PR #472, which is a really small one-line change, and merge it if you are satisfied. 

(After that there is just one more small issue [cf-conventions/#516](https://github.com/cf-convention/cf-conventions/issues/516) to close before the main act in #470, which has no PR at the moment.)

Many thanks,
Lars

**Comment by sadielbartholomew** [comment]
Sure, thanks for raising that PR Lars. Since it is small to review I will do it now.
